cilium-cli/install/autodetect.go

```diff
@@ -87,7 +101,12 @@ func (k *K8sInstaller) autodetectAndValidate(ctx context.Context, helmValues map
 	if k.params.ClusterName == "" {
 		if k.flavor.ClusterName != "" {
 			// Neither underscores nor dots are allowed as part of the cluster name.
-			name := strings.NewReplacer("_", "-", ".", "-").Replace(k.flavor.ClusterName)
+			name := strings.NewReplacer("_", "-", ".", "-", ":", "-").Replace(k.flavor.ClusterName)
+
+			if k.flavor.Kind == k8s.KindEKS {
+				name = trimEKSClusterARN(k.flavor.ClusterName)
```

Comment:
Don't we want to trim the fixed-up name here?

```suggestion
				name = trimEKSClusterARN(name)
```

Reply:
@tklauser I have implemented the changes you suggested. Thank you!

Reply:
Thanks!